### PR TITLE
Make Tickets page and Somewhat working Edit Ticket page

### PR DESCRIPTION
**Pull request by Val-Bustamante**
Make Ticket page works (interface and functionality). The info button for categories is ugly so I'll make it better lookin down the road. 

Edit Ticket page has an interface and a button in View Ticket that lets you access the page. Actual functionality though does not work yet.

### CODE REVIEW

File: views/editTicket.handlebars [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    var temp2 = "{{category}}";
+    var mySelect2 = document.getElementById('ticketCategory');
+
+    for(var i, j = 0; i = mySelect2.options[j]; j++) {
+        if(i.value == temp2) {
+            mySelect2.selectedIndex = j;
+            break;
+        }
+    }
+</script>
```
Comment (MAPReiff) [inline views/editTicket.handlebars]: I believe the prof said we can not use inline JS - please move to a dedicated file

File: data/tickets.js [hunk 2]
```diff
@@ -27,18 +33,19 @@ const create = async (
     status != "In Progress" &&
     status != "Completed"
   ) {
-    throw new Error("status must be a string equal to, To Do, In Progress, or Completed");
+    throw new Error("status must be a string equal to To Do, In Progress, or Completed");
   }
 
   // validate priority
   priority = helpers.checkString(priority, "Priority");
+  console.log(priority);
```
Comment (MAPReiff) [inline data/tickets.js]: Please remove unnecessary console.logs

File: public/js/makeTicket.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+
+console.log("in here")
+var form = document.getElementById("makeTicketForm");
+if (form) {
+    console.log("wow");
```
Comment (MAPReiff) [inline public/js/makeTicket.js]: Please remove unnecessary console.logs

File: routes/tickets.js [hunk 4]
```diff
@@ -93,4 +94,138 @@ router
...
+      res.status(200).render("makeTicket", {
+        title: "Create Ticket"
+      });
+    } catch (e) {
+      renderError(res, 500, 'Internal Server Error');
+    }
+    
+    //code here for POST
+  }).post(async (req, res) => {
+    console.log("IN MAKE TICKET POST")
```
Comment (MAPReiff) [inline routes/tickets.js]: Please remove unnecessary console.logs

File: routes/tickets.js [hunk 5]
```diff
@@ -93,4 +94,138 @@ router
...
+            ticketPriority,
+            req.body["ticketDeadline"],
+            req.session.user._id,
+            ['644939d4475bc92a43a50aef'],
+            ticketCategory
+            //[]
+          );
+
+          if (createdTicket) {
+            console.log("ticketId",createdTicket._id);
```
Comment (MAPReiff) [inline routes/tickets.js]: Please remove unnecessary console.logs

**Review by MAPReiff — CHANGES_REQUESTED** [review]
A few comments left in files, please correct them when you get a chance.

Otherwise looks good.

**Comment by MAPReiff** [comment]
@Val-Bustamante bumping this regarding the comment on the script in `editTicket.handlebars`

**Review by MAPReiff — APPROVED** [review]

File: routes/tickets.js [hunk 6]
```diff
@@ -93,4 +94,136 @@ router
...
+    } catch (e) {
+      renderError(res, 500, 'Internal Server Error');
+    }
+    
+    //code here for GET
+  }).post(async (req, res) => {
+    let ticket;
+
+    try {
+      ticket = await ticketData.get(req.params.id);
```
Comment (bigguyhere) [inline routes/tickets.js:154]: Ticket is assigned a value, but never used

File: routes/tickets.js [hunk 7]
```diff
@@ -93,4 +94,136 @@ router
...
+      renderError(res, 500, 'Internal Server Error');
+    }
+    
+    //code here for POST
+  }).post(async (req, res) => {
+
+    let user;
+
+    try {
+      user = await userData.get(req.session.user._id);
```
Comment (bigguyhere) [inline routes/tickets.js:182]: User is assigned a value, but never used

**Pull request merged**